src/session.ts

```diff
@@ -588,24 +589,25 @@ async function sendReinvite(this: WebPhoneSession, options: any = {}): Promise<a
     this.pendingReinvite = true;
     options.modifiers = options.modifiers || [];
 
-    return new Promise(async (resolve, reject) => {
-        try {
-            const description = await this.sessionDescriptionHandler.getDescription(
-                options.sessionDescriptionHandlerOptions,
-                options.modifiers
-            );
+    try {
+        const description = await this.sessionDescriptionHandler.getDescription(
+            options.sessionDescriptionHandlerOptions,
+            options.modifiers
+        );
+        const result = await new Promise((resolve, reject) => {
             this.sendRequest(C.INVITE, {
                 body: description,
                 receiveResponse: (response: IncomingResponse) => {
                     if (response.statusCode === 200) resolve(response);
-                    return this.receiveReinviteResponse(response);
                 }
             });
-        } catch (e) {
-            this.pendingReinvite = false;
-            reject(e);
-        }
-    });
+        });
+        await this.receiveReinviteResponse(result);
+        return result;
+    } catch (e) {
+        this.pendingReinvite = false;
+        throw new Error('Reinvite Failed with the reason ' + e);
```

Comment:
`e` is type `Error`, it would be converted to string but better either just `throw e` or `throw new Error('Reinvite Failed with the reason ' + e.message);`

Reply:
### I tot i could capure the message , but should be pretty straightforward to debug the issue with sip msgs.  I will change it to  e.message